BeeSwift/AddDataIntentHandler.swift

```diff
@@ -12,28 +12,46 @@ import BeeKit
 
 @available(iOS 14.0, *)
 class AddDataIntentHandler: NSObject, AddDataIntentHandling {
-    func resolveValue(for intent: AddDataIntent, with completion: @escaping (AddDataValueResolutionResult) -> Void) {
-        print("foo")
+    func resolveValue(for intent: AddDataIntent) async -> AddDataValueResolutionResult {
+        if let value = intent.value {
+            return AddDataValueResolutionResult.success(with: value.doubleValue)
+        } else {
+            return AddDataValueResolutionResult.needsValue()
+        }
     }
     
-    func resolveGoal(for intent: AddDataIntent, with completion: @escaping (INStringResolutionResult) -> Void) {
-        print("foo")
+    func resolveGoal(for intent: AddDataIntent) async -> INStringResolutionResult {
+        if let goal = intent.goal {
+            // TODO: We should validate this is a valid slug
+            return INStringResolutionResult.success(with: goal)
+        } else {
+            return INStringResolutionResult.needsValue()
+        }
     }
     
-    func confirm(intent: AddDataIntent,
-               completion: @escaping (AddDataIntentResponse) -> Void) {
-        completion(AddDataIntentResponse(code: .ready, userActivity: nil))
+    func confirm(intent: AddDataIntent) async -> AddDataIntentResponse {
+        if intent.goal != nil && intent.value != nil {
+            return AddDataIntentResponse(code: .ready, userActivity: nil)
+        } else {
+            return AddDataIntentResponse(code: .failure, userActivity: nil)
+        }
     }
 
     func handle(intent: AddDataIntent,
               completion: @escaping (AddDataIntentResponse) -> Void) {
-        guard let datapointValue = intent.value else { return }
-        
-        RequestManager.addDatapoint(urtext: "^ \(datapointValue)", slug: intent.goal!) { (response) in
-            completion(AddDataIntentResponse.success(goal: intent.goal!))
+        guard let goal = intent.goal else {
```

Comment:
How about naming this variable `goalSlug` or `slug`? It is not a full JSONGoal but merely a String containing the goal's slug.

Reply:
This seems reasonable. In a perfect world we would rename the property on the Intent to `goalSlug` and then do `guard let goalSlug = intent.goalSlug ...`. However, given the intent already exists, renaming the property would break existing shortcuts which I'm reluctant to do.

This means the best we can get to is `guard let goalSlug = intent.goal ...` - trading off a better name for less consistency. So not a costless win, but maybe still net better?

Reply:
Here I only meant the new code which introduced the `guard let goal = ...` variable. That local variable might better named slug.